lib/rubocop/cop/style/auto_resource_cleanup.rb

```diff
@@ -23,21 +23,34 @@ class AutoResourceCleanup < Cop
           File: :open
         }.freeze
 
+        CLEANUP_METHODS = {
+          File: :close
+        }.freeze
+
         def on_send(node)
           TARGET_METHODS.each do |target_class, target_method|
             target_receiver = s(:const, nil, target_class)
 
             next if node.receiver != target_receiver
             next if node.method_name != target_method
-            next if node.parent && node.parent.block_type?
-            next if node.block_argument?
+            next if cleanup?(node, target_class)
 
             add_offense(node,
                         message: format(MSG,
                                         class: target_class,
                                         method: target_method))
           end
         end
+
+        private
+
+        def cleanup?(node, target_class)
+          parent = node.parent
+          cleanup_method = CLEANUP_METHODS[target_class]
+          return true if parent && parent.block_type?
```

Comment:
What are your thoughts on combining this into a single statement?

```ruby
node.block_argument? ||
  (parent && (parent.block_type? || parent.method_name == cleanup_method))
```

Reply:
:+1: